backend/src/main/java/com/github/NicolaiKopka/users/AccountService.java

```diff
@@ -0,0 +1,36 @@
+package com.github.NicolaiKopka.users;
+
+import com.github.NicolaiKopka.dto.RegisterUserDTO;
+import com.github.NicolaiKopka.users.RegisterData;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class AccountService {
+
+    private final MyUserRepo myUserRepo;
+    private final PasswordEncoder encoder;
+    public MyUser registerUser(RegisterData registerData) {
+
+        if(registerData.getUsername().isBlank() || registerData.getPassword().isBlank() || registerData.getCheckPassword().isBlank()) {
+            throw new IllegalArgumentException("No empty fields allowed");
+        }
+        if(myUserRepo.findByUsername(registerData.getUsername()).isPresent()) {
```

Comment:
If two users try to create an user with the same username at the same time, both would end up in the database. Add a unique index for the field username in your `MyUser` class.